compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1356,6 +1356,14 @@ impl<'tcx> TyCtxt<'tcx> {
         self.diagnostic_items(did.krate).name_to_id.get(&name) == Some(&did)
     }
 
+    /// Returns `Some(async_generator_kind)` if the node pointed to by `def_id` is an async generator.
+    pub fn async_kind(def_id: DefId) -> Option<hir::AsyncGeneratorKind> {
```

Comment:
all the usages of this function are currently just checking `.is_some()`, is there a reason fr not returning a bool?

Reply:
Indeed. Looks like all the cases where the async kind itself is checked are using the hir body directly, and not through `TyCtxt`.